src/containers/EnterpriseApp/index.jsx

```diff
@@ -9,17 +9,38 @@ import AdminPage from '../AdminPage';
 import NotFoundPage from '../NotFoundPage';
 import ErrorPage from '../ErrorPage';
 
-import { fetchPortalConfiguration } from '../../data/actions/portalConfiguration';
+import fetchEnterpriseList from '../../data/actions/enterpriseList';
+import { setPortalConfiguration } from '../../data/actions/portalConfiguration';
 import { getLocalUser } from '../../data/actions/authentication';
 
 class EnterpriseApp extends React.Component {
   componentDidMount() {
     const { enterpriseSlug } = this.props.match.params;
+    if (!this.props.enterprises) {
+      this.props.fetchEnterpriseList();
+    } else {
+      const enterprise = this.findEnterprise(enterpriseSlug);
+      this.props.setPortalConfiguration(enterprise);
+    }
 
-    this.props.getPortalConfiguration(enterpriseSlug);
     this.props.getLocalUser();
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
The logic here handles the case when the enterprises already exist (i.e., user coming from the `EnterpriseList` page), so we just need to dispatch the `setPortalConfiguration` action.